src/amo/components/Collection/index.js

```diff
@@ -127,7 +128,7 @@ export class CollectionBase extends React.Component<Props> {
       return;
     }
 
-    if (collection && addonsPageChanged) {
+    if (collection && addonsPageChanged && collection.numberOfAddons) {
```

Comment:
Sorry but why do we need this change?

Reply:
Because we don't want to fetch the collection page (which fetches the add-ons) for a collection without any add-ons. When we redirect to the collection edit page after creating a new collection, fetching the add-ons causes a flash of content because it thinks we're waiting to load add-ons.